src/project-css/app/Adapters/ImportAdapter.php

```diff
@@ -0,0 +1,208 @@
+     * merge the 2 lines.
+     * @param array $tkns1
+     * @param array $tkns2
+     * @return array
+     */    
+    public function mergeLines() {
+        $numItem = count($this->savedTkns) - 1;
+        $this->savedTkns[ $numItem ] = $this->savedTkns[ $numItem ] . ' ' . $this->tkns[ 0 ];
+        unset($this->tkns[ 0 ]);
+        $this->tkns = ( (count($this->tkns) > 0) ? array_merge($this->savedTkns, $this->tkns) : $this->savedTkns );
+    }
+
+    /**
+     * takes a string and prepares it to be used inserted as a new record
+     * if we do not find enough items in the line we will check and see if
+     * the previous line was saved and merge them and check the count again.
+     * if their is insufficent items we will save the tkns and the row position
+     * so we can attempt a merge later.
+     * @param string $curLine current line read from the file
+     * @param string $delimiter type of delmiter that is used in the file
+     * @param integer $orgCount current number of fields expected
+     * @param integer $prcssd current position in the file
+     * @return string
+     */
+    public function prepareLine($curLine, $delimiter, $orgCount, $prcssd) {
+        // Strip out Quotes that are sometimes seen in csv files around each item
+        $curLine = str_replace('"', "", $curLine);
+
+        // Tokenize the line
+        $this->tkns = (new CSVHelper)->tknze($curLine, $delimiter);
+
+        // Validate the tokens and filter them
+        $this->tkns = (new CSVHelper)->fltrTkns($this->tkns);
+
+        // if lastErrRow is the previous row try to combine the lines
+        if ((count($this->tkns) != $orgCount) && ($this->lastErrRow == $prcssd) && ($this->savedTkns != NULL)) {
+            $this->mergeLines();
+
+            // clear last saved line since we did a merge
+            $this->lastErrRow = NULL;
+            $this->savedTkns = NULL;
+        }
+
+        // if token count doesn't match what is exptected save the tkns and last row position
+        if (is_array($this->tkns) && (count($this->tkns) != $orgCount)) {
+          // save the last row position and the Tokenized row
+          $this->lastErrRow = $prcssd;
+          $this->savedTkns = $this->tkns;
+          return false;
+        }
+
+        return true;
+    }
+
+    /**
+    * takes array of tokens. Creates a search index and 
+    * inserts them into the table
+    * 
+    * @param array $tkns array containing all fields of record
+    * @param integer $orgCount number of expected fields in the record
+    * @param object $clmnLst array containing all field names for table
+    * @return boolean
+    */   
+    public function processLine($orgCount) {
+        if(!is_array($this->tkns) && empty($this->tkns)) { return false; }
+
+        // verify that passed $tkns match the expected field count
+        if (count($this->tkns) == $orgCount) {
+          // Declae an array
+          $this->curArry = array();
```

Comment:
Why are you setting so many instance variables in the class is there a reason?  Is this to pass data from function to function without using the params?

Reply:
Yea I was using alot of class instance variables in the adapter. After looking at your comments I think I can set this up better so it's more readable.